internal/core/data/init.go

```diff
@@ -48,6 +50,13 @@ func (b *Bootstrap) BootstrapHandler(ctx context.Context, wg *sync.WaitGroup, st
 	configuration := dataContainer.ConfigurationFrom(dic.Get)
 	lc := container.LoggingClientFrom(dic.Get)
 
+	timeout, err := time.ParseDuration(configuration.Service.RequestTimeout)
+	if err != nil {
+		lc.Errorf("unable to parse RequestTimeout value of %s to a duration: %w", configuration.Service.RequestTimeout, err)
+		return false
+	}
+	b.router.Use(correlation.ManageTimeout(timeout))
+
```

Comment:
All edgex-go services need this, so it should be common func that takes RequestTimeout and router as params.

Reply:
Added the common func.

Reply:
Created PR to remove the timeout setting from bootstrap lib https://github.com/edgexfoundry/go-mod-bootstrap/pull/267

Reply:
Yes, I think that would be good as it would then provide the capability to all services.

Reply:
Moved the timeout handler to go-mod-bootstrap by PR https://github.com/edgexfoundry/go-mod-bootstrap/pull/267 and close this PR.